app/nasa_collaboration/page.js

```diff
@@ -1,31 +1,49 @@
 "use client";
 
-import React, { useState, useEffect } from 'react';
-import styles from './page.module.css';
+import React, { useState, useEffect } from "react";
+import styles from "./page.module.css";
 
-// Read "/app/nasa_collaboration/README.md" for more info about the API_KEY
-// You need a proper API_KEY for the requests to work
-const API_KEY = 'API_KEY';
+const API_KEY = "XM2aaPe8YoOGcLyFP55Qg5TxfJAhIW3CrEZEeh9H";
 
 const NASA_URLs = {
   astronomyPicOfTheDay: `https://api.nasa.gov/planetary/apod?api_key=${API_KEY}`,
   marsRoverPhoto: `https://api.nasa.gov/mars-photos/api/v1/rovers/curiosity/photos?earth_date=2015-6-3&api_key=${API_KEY}`,
 };
 
+const RoverPhoto = ({ src, date, roverName }) => {
+  return (
+    <div className={styles.nasaPicOfTheDayImg}>
+      <img src={src} alt={`Rover: ${roverName} on ${date}`} />
+      <p className={styles.roverText}>{`Date: ${date}`}</p>
+      <p className={styles.roverText}>{`Rover: ${roverName}`}</p>
+    </div>
+  );
+};
+
 export const NasaCollaboration = () => {
   const [dailyImg, setDailyImg] = useState({});
   const [roverPhoto, setRoverPhoto] = useState({});
 
   useEffect(() => {
     const fetchRoverPhotos = async () => {
-      const roverPhotoResponse = await fetch(NASA_URLs.marsRoverPhoto).then(response => response.json());
+      const roverPhotoResponse = await fetch(NASA_URLs.marsRoverPhoto).then(
+        (response) => response.json()
+      );
       setRoverPhoto(roverPhotoResponse);
     };
 
-    fetchRoverPhotos();
+    const fetchData = async () => {
+      try {
+        const dailyImgResponse = await fetch(NASA_URLs.astronomyPicOfTheDay);
+        const dailyImg = await dailyImgResponse.json();
+        setDailyImg(dailyImg);
+      } catch (error) {
+        console.error("Error fetching astronomy picture of the day:", error);
```

Comment:
How is this error communicated to the user? How will the user know something went wrong? :wink: